tests/psi4numpy/rhf_gradient/input.py

```diff
@@ -0,0 +1,163 @@
+import time
+import numpy as np
+np.set_printoptions(precision=15, linewidth=200, suppress=True)
+import psi4
+
+def rhf_gradient(mol, options):
+
+    psi4.core.set_active_molecule(mol)
+    psi4.set_options(options)
+
+    rhf_e, wfn = psi4.energy('SCF', return_wfn=True)
+
+    # Assuming C1 symmetry    
+    occ = wfn.doccpi()[0]
+    nmo = wfn.nmo()
+
+    C = wfn.Ca()
+    npC = np.asarray(C)
+
+    mints = psi4.core.MintsHelper(wfn.basisset())
+    H_ao = np.asarray(mints.ao_kinetic()) + np.asarray(mints.ao_potential())
+
+    # Update H, transform to MO basis 
+    H = np.einsum('uj,vi,uv', npC, npC, H_ao)
+
+    # Integral generation from Psi4's MintsHelper
+    MO = np.asarray(mints.mo_eri(C, C, C, C))
+    # Physicist notation    
+    MO = MO.swapaxes(1,2)
+
+    F = H + 2.0 * np.einsum('pmqm->pq', MO[:, :occ, :, :occ])
+    F -= np.einsum('pmmq->pq', MO[:, :occ, :occ, :])
+    natoms = mol.natom()
+    cart = {}; cart[0] = '_X'; cart[1] = '_Y'; cart[2] = '_Z'
```

Comment:
It looks like you are building a set here? `{'_x', '_y', '_z'}` will do the same thing.

Reply:
{} creates a dict, not a set.

On that subject, might I suggest using a tuple here? When all you need is to access elements based on an index, a tuple will suffice and is more Pythonic.

Reply:
Ah, yes. Too early in the morning. Either way, best to do it as an explicit build.

Reply:
I think a list is the best way to go in my case as set doesn't support indexing and a tuple might be unnecessary for this simple example.